src/Controller/UserController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Controller;
+
+use App\Controller\Dto\UserDto;
+use App\Entity\User;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Validator\ConstraintViolation;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * @Route (path="/api/user")
+ */
+class UserController
+{
+    private EntityManagerInterface $entityManager;
+
+    private ValidatorInterface $validator;
+
+    public function __construct(EntityManagerInterface $entityManager, ValidatorInterface $validator)
+    {
+        $this->entityManager = $entityManager;
+        $this->validator = $validator;
+    }
+
+    /**
+     * @Route(methods={"POST"})
+     */
+    public function register(UserDto $userDto): Response
+    {
+        $user = User::createFromDto($userDto);
+        $errors = $this->validator->validate($user);
+
+        if (count($errors) > 0) {
+            $errorArray = [];
+            foreach ($errors as $error) {
+                /**
+                 * @var ConstraintViolation $error
```

Comment:
this kind of annotations (@var for an element from a foreach) should go above the foreach and can be wrriten on a single line (i.e. `/** @var Class $object */`)

Reply:
ok